src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -113,10 +138,24 @@ public String getProcess() {
         return promotionProcessName;
     }
 
+    @Nonnull
+    @Deprecated
     public List getBuilds() {
-        List builds = new ArrayList();
+        // Try to get ancestor from the object
+        Item item = Stapler.getCurrentRequest().findAncestorObject(Item.class);
+        return getBuilds(item);
+    }
+    
+    @Nonnull
+    public List getBuilds(@CheckForNull Item base) {
+        final List builds = new ArrayList();
+        final Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins == null) {
+            return builds;
+        }
 
-        AbstractProject job = (AbstractProject) Jenkins.getInstance().getItem(projectName);
+        // JENKINS-25011: also look for jobs in folders.
+        final AbstractProject job = getItemByPath(projectName, base, AbstractProject.class);
```

Comment:
Yes, the plugin does not support workflow & Co now. IMO it's another story to be done separately.